drivers/pci/controller/vmd.c

```diff
@@ -553,8 +563,10 @@ static void vmd_domain_reset(struct vmd_dev *vmd)
 
 static void vmd_attach_resources(struct vmd_dev *vmd)
 {
-	vmd->dev->resource[VMD_MEMBAR1].child = &vmd->resources[1];
-	vmd->dev->resource[VMD_MEMBAR2].child = &vmd->resources[2];
+	vmd->dev->resource[VMD_MEMBAR1].child =
+		&vmd->resources[VMD_RESOURCE_MEMBAR_1];
```

Comment:
For sure, it is badly formatted. 

```suggestion
struct x *vmd_membar1_res=&vmd->dev->resource[VMD_MEMBAR1];
struct x *vmd_membar2_res =&vmd->dev->resource[VMD_MEMBAR2];

vmd_membar1_res->child =&vmd->dev->resource[VMD_MEMBAR1];
vmd_membar2_res->child =&vmd->dev->resource[VMD_MEMBAR2];
```